### PR TITLE
DEV-2330 split of AdjustableCurrentController

### PR DESCRIPTION
**Pull request by EdytaMalcharek**

### CODE REVIEW

File: adapter/service/chargepoint/service.go [hunk 1]
```diff
@@ -363,21 +370,43 @@ func (s *service) SupportedStates() []string {
...
 	return err == nil
 }
 
-// adjustableCurrentController returns the AdjustableCurrentController, if supported.
-func (s *service) adjustableCurrentController() (AdjustableCurrentController, error) {
+// adjustableMaxCurrentController returns the AdjustableMaxCurrentController, if supported.
+func (s *service) adjustableMaxCurrentController() (AdjustableMaxCurrentController, error) {
+	_, ok := s.Specification().PropertyInteger(PropertySupportedMaxCurrent)
+	if !ok {
+		return nil, fmt.Errorf("%s: adjusting current is not supported", s.Name())
```
Comment (rafalmnich) [inline adapter/service/chargepoint/service.go:391]: Adjust error strings according to property names.

File: adapter/service/chargepoint/service.go [hunk 2]
```diff
@@ -363,21 +370,43 @@ func (s *service) SupportedStates() []string {
...
+	}
+
+	return controller, nil
+}
+
+// adjustableOfferedCurrentController returns the AdjustableOfferedCurrentController, if supported.
+func (s *service) adjustableOfferedCurrentController() (AdjustableOfferedCurrentController, error) {
 	_, ok := s.Specification().PropertyInteger(PropertySupportedMaxCurrent)
```
Comment (rafalmnich) [inline adapter/service/chargepoint/service.go:404]: `PropertyOfferedCurrent` ?

**Review by rafalmnich — CHANGES_REQUESTED** [review]
Pls also rebase / merge with main branch, to resolve conflicts.

**Review by rafalmnich — APPROVED** [review]

File: adapter/service/chargepoint/tasks.go [hunk 3]
```diff
@@ -25,6 +25,7 @@ func handleReporting(serviceRegistry adapter.ServiceRegistry) func() {
 				continue
 			}
 
+<<<<<<< HEAD
```
Comment (rafalmnich) [inline adapter/service/chargepoint/tasks.go:28]: You haven't finished resolving the conflicts, as the conflicts markers are still here.

**Review by rafalmnich — CHANGES_REQUESTED** [review]

File: adapter/service/chargepoint/service.go [hunk 4]
```diff
@@ -85,8 +88,8 @@ type Service interface {
 	SendPhaseModeReport(force bool) (bool, error)
 	// SupportedStates returns states that are supported by the chargepoint.
 	SupportedStates() []string
-	// SupportsAdjustingCurrent returns true if the chargepoint supports adjusting current.
-	SupportsAdjustingCurrent() bool
+	// SupportsAdjustingMaxCurrent returns true if the chargepoint supports adjusting current.
+	SupportsAdjustingMaxCurrent() bool
```
Comment (rafalmnich) [inline adapter/service/chargepoint/service.go:92]: I think that it should check if supports offered current, and maybe in addition - the max current. @boguslaw-wojcik what do you think?

```suggestion
	// SupportsAdjustingOfferedCurrent returns true if the chargepoint supports adjusting offered current.
	SupportsAdjustingOfferedCurrent() bool
```

**Review by rafalmnich — CHANGES_REQUESTED** [review]

File: adapter/service/chargepoint/service.go [hunk 5]
```diff
@@ -456,16 +470,31 @@ func (s *service) SupportsAdjustingPhaseModes() bool {
...
+	if !ok {
+		return nil, fmt.Errorf("%s: adjusting max current is not supported", s.Name())
+	}
+
+	return controller, nil
+}
+
+// adjustableOfferedCurrentController returns the AdjustableOfferedCurrentController, if supported.
+func (s *service) adjustableOfferedCurrentController() (AdjustableOfferedCurrentController, error) {
+	_, ok := s.Specification().PropertyInteger(PropertyOfferedCurrent)
```
Comment (boguslaw-wojcik) [inline adapter/service/chargepoint/service.go:490]: Nope, this is incorrect. PropertyOfferedCurrent is a property of "evt.current_session.report" interface, not a service specification.

What you should check here, is if there is sup_max_current property defining an upper boundary for setting offered current.

**Review by boguslaw-wojcik — DISMISSED** [review]
As per left comments.

File: adapter/service/chargepoint/service.go (hunk 5 above)
Comment (boguslaw-wojcik) [reply adapter/service/chargepoint/service.go:490]: The logic dictates we have this requirement for both of the controllers.

File: adapter/service/chargepoint/service.go (hunk 2 above)
Comment (boguslaw-wojcik) [reply adapter/service/chargepoint/service.go:490]: Nope.

**Review by boguslaw-wojcik — APPROVED** [review]

**Pull request merged**